src/commands/send_event.rs

```diff
@@ -174,12 +183,33 @@ pub fn execute(matches: &ArgMatches) -> Result<()> {
         }
 
         for path in collected_paths {
-            let p = path.as_path();
-            let file = File::open(p)?;
-            let reader = BufReader::new(file);
-            let event: Event = serde_json::from_reader(reader)?;
-            let id = send_raw_event(event, dsn.clone());
-            println!("Event from file {} dispatched: {}", p.display(), id);
+            let raw_event = std::fs::read(&path)?;
+
+            let id = if raw {
+                use std::io::Write;
+
+                // Its a bit unfortunate that we still need to parse the whole JSON,
+                // but envelopes need an `event_id`, which we also want to report.
+                let json: Value = serde_json::from_slice(&raw_event)?;
+                let id = json
+                    .as_object()
+                    .and_then(|event| event.get("event_id"))
+                    .and_then(|val| val.as_str())
+                    .and_then(|id| Uuid::parse_str(id).ok())
+                    .unwrap_or_default();
+                let mut buf = Vec::new();
+                writeln!(buf, r#"{{"event_id":"{id}"}}"#)?;
+                writeln!(buf, r#"{{"type":"event","length":{}}}"#, raw_event.len())?;
```

Comment:
Wouldn't this write two separate objects with one field each to the buffer? Is that intended?

Reply:
Yes, the first is the envelope header with the `event_id`, the other one is the item header with type/len.

Reply:
Doesn't that mean the contents of the envelope won't be a valid json value?

Reply:
An envelope can have multiple JSON payloads or even binary files in one container, so it is not valid JSON *as a whole*.

Reply:
Right, that explains it. Thanks!